modules/consensus/subscribe.go

```diff
@@ -165,16 +168,23 @@ func (cs *ConsensusSet) managedInitializeSubscribe(subscriber modules.ConsensusS
 	})
 	cs.mu.RUnlock()
 	if err != nil {
-		return err
+		return modules.ConsensusChangeID{}, err
+	}
+
+	// Nothing to do if the changeEntry doesn't exist.
+	if !exists {
+		return start, nil
```

Comment:
is this code block reachable? Looks to me like it returns inside the loop if !exists?

Reply:
I think so. It's not part of the loop. In line `165` we set `entry, exists = entry.NextEntry(tx)` to get the first entry that we have to send to the new subscriber. If that new entry doesn't exist we return early.

Reply:
ah I see